src/utils.c

```diff
@@ -424,3 +424,100 @@ SEXP startsWithAny(const SEXP x, const SEXP y, SEXP start) {
   return ScalarLogical(false);
 }
 
+SEXP frev(SEXP x, SEXP copyArg) {
+  if (INHERITS(x, char_dataframe) || INHERITS(x, char_datatable))
+    error(_("'x' should not be data.frame or data.table."));
+  if (!isNull(getAttrib(x, R_DimSymbol)))
+    error(_("'x' should not be matrix or array"));
+  if (!IS_TRUE_OR_FALSE(copyArg))
+    error(_("%s must be TRUE or FALSE."), "copy");
+  bool copy = LOGICAL(copyArg)[0];
+  R_xlen_t n = xlength(x);
+  int nprotect = 0;
+  if (copy) {
+    x = PROTECT(duplicate(x));
+    nprotect++;
```

Comment:
IINM there's only one `PROTECT()` here; the `nprotect` seems like overkill. `UNPROTECT(1)` should do fine.

Reply:
Maybe even `UNPROTECT(copy)` to cater for the copy/non-copy cases?

Reply:
I think I'd still prefer `UNPROTECT(1)`, memory safety is hard enough to keep track of mentally that it's nice to have a literal value we can use here -- having to mentally map `copy == FALSE` --> `UNPROTECT(0)`, `copy = TRUE` --> `UNPROTECT(1)` is a tax on the reader IMO

Reply:
I guess I'm on the wrong track here, but are you suggesting:
```r
  if (copy) {
    x = PROTECT(duplicate(x));
    UNPROTECT(1);
  }
```
and remove the other UNPROTECT calls? I thought that's invalid because we modify x afterwards?

Reply:
I see what you mean now, sorry, `UNPROTECT(copy)` works for me, otherwise the current `nprotect` is also OK.